yarn-project/circuits.js/src/contract/contract_class_id.ts

```diff
@@ -56,8 +59,17 @@ export type ContractClassIdPreimage = {
   publicBytecodeCommitment: Fr;
 };
 
-// TODO(#5860): Replace with actual implementation
-// Changed to work with canonical contracts that may have non-deterministic noir compiles and we want to keep the address constant
-export function computePublicBytecodeCommitment(_bytecode: Buffer) {
-  return new Fr(5);
+export function computePublicBytecodeCommitment(packedBytecode: Buffer) {
+  // Encode the buffer into field elements (chunked into 31 bytes each)
+  const encodedBytecode: Fr[] = bufferAsFields(packedBytecode, MAX_PACKED_PUBLIC_BYTECODE_SIZE_IN_FIELDS);
+  // The first element is the length of the bytecode (in bytes)
```

Comment:
Is this still the case? Given that a buffer holds its length, I thought I would've avoided this extra byte in the packedBytecode.

You can however add it here in this function if you need it (but not in the function that generates the packedBytecode).

Reply:
> You can however add it here in this function 
I meant in the function that passes the bytecode to the contract registerer

Here you should be able to use the buffer size since we are in TS?

Reply:
Yeah that's true, I was just re-using the `bufferAsFields` utility since that's what's used when broadcasting public functions and i didnt want to re-implement the "packing into fields" logic.

I could remove the length from the encoded-bytecode that is used in `pop_capsule`, but we would need to then add it as a parameter to the class register function.